### PR TITLE
update HTTP caching doc

### PR DESCRIPTION
**Pull request by senmu-a**
update HTTP caching doc

### CODE REVIEW

File: files/zh-cn/web/http/caching/index.md [hunk 1]
```diff
@@ -5,154 +5,663 @@ original_slug: Web/HTTP/Caching_FAQ
...
 
-{{HTTPHeader("Last-Modified")}} 响应头可以作为一种弱校验器。说它弱是因为它只能精确到一秒。如果响应头里含有这个信息，客户端可以在后续的请求中带上 {{HTTPHeader("If-Modified-Since")}} 来验证缓存。
+请注意，数字 “41” 具有最长的 `max-age`（1 年），但具有 `public`。
 
-当向服务端发起缓存校验的请求时，服务端会返回 {{HTTPStatus(200)}} ok 表示返回正常的结果或者 {{HTTPStatus(304)}} Not Modified(不返回 body) 表示浏览器可以使用本地缓存文件。304 的响应头也可以同时更新缓存文档的过期时间。
+`public` 值具有使响应可存储的效果，即使存在 `Authorization` 标头。
 
-## Vary 响应
+> **注意：** 只有在设置了 `Authorization` 标头时需要存储响应时才应使用 `public` 指令。
+> 否则不需要，因为只要给出了 `max-age`，响应就会存储在共享缓存中。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:556]:
```suggestion
> **备注：** 只有在设置了 `Authorization` 标头时需要存储响应时才应使用 `public` 指令。否则不需要，因为只要给出了 `max-age`，响应就会存储在共享缓存中。
```

File: files/zh-cn/web/http/caching/index.md [hunk 2]
```diff
@@ -5,154 +5,663 @@ original_slug: Web/HTTP/Caching_FAQ
...
+ETag: YsAIAAAA-QG4G6kCMAMBAAAAAAAoK
+```
 
-![The Vary header leads cache to use more HTTP headers as key for the cache.](https://mdn.mozillademos.org/files/13769/HTTPVary.png)
+**缓存破坏** 是一种通过在内容更改时更改 URL 来使响应在很长一段时间内可缓存的技术。该技术可以应用于所有子资源，例如图像。
 
-使用 vary 头有利于内容服务的动态多样性。例如，使用 Vary: User-Agent 头，缓存服务器需要通过 UA 判断是否使用缓存的页面。如果需要区分移动端和桌面端的展示内容，利用这种方式就能避免在不同的终端展示错误的布局。另外，它可以帮助 Google 或者其他搜索引擎更好地发现页面的移动版本，并且告诉搜索引擎没有引入[Cloaking](https://en.wikipedia.org/wiki/Cloaking)。
+> **注意：** 在评估 `immutable` 和 QPACK 的使用时：
+> 如果你担心 `immutable` 会更改 QPACK 提供的预定义值，请考虑
+> 在这种情况下，`immutable` 部分可以通过将 `Cache-Control` 值分成两行来单独编码——尽管这取决于特定 QPACK 实现使用的编码算法。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:600]:
```suggestion
> **备注：** 在评估 `immutable` 和 QPACK 的使用时：如果你担心 `immutable` 会更改 QPACK 提供的预定义值，请考虑在这种情况下，`immutable` 部分可以通过将 `Cache-Control` 值分成两行来单独编码——尽管这取决于特定 QPACK 实现使用的编码算法。
```

File: files/zh-cn/web/http/caching/index.md [hunk 3]
```diff
@@ -5,154 +5,663 @@ original_slug: Web/HTTP/Caching_FAQ
...
+
+大多数 Web 内容都可以通过上述两种模式的组合来覆盖。
+
+### 有关托管缓存的更多信息
+
+使用前面章节描述的方法，子资源可以通过缓存破坏来缓存很长时间，但主资源（通常是HTML文档）不能。
+
+缓存主要资源很困难，因为仅使用 HTTP 缓存规范中的标准指令，在服务器上更新内容时无法主动删除缓存内容。
+
+但是，可以通过部署托管缓存（例如 CDN 或服务工作者）来实现。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:656]:
```suggestion
但是，可以通过部署托管缓存（例如 CDN 或 service worker）来实现。
```

File: files/zh-cn/web/http/caching/index.md [hunk 4]
```diff
@@ -5,154 +5,663 @@ original_slug: Web/HTTP/Caching_FAQ
...
+
+例如，允许通过 API 或仪表板操作清除缓存的 CDN 将通过存储主要资源并仅在服务器上发生更新时显式清除相关缓存来实现更积极的缓存策略。
+
+如果服务工作者可以在服务器上发生更新时删除缓存 API 中的内容，它也可以这样做。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:660]:
```suggestion
如果 service worker 可以在服务器上发生更新时删除缓存 API 中的内容，它也可以这样做。
```

File: files/zh-cn/web/http/caching/index.md [hunk 5]
```diff
@@ -5,154 +5,663 @@ original_slug: Web/HTTP/Caching_FAQ
...
+
+有关更多信息，请参阅 CDN 文档，并查阅 [service worker 文档](/zh-CN/docs/Web/API/Service_Worker_API)。
 
 ## 参考
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:664]:
```suggestion
## 参见
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions, you can refer: <https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md>

Thanks for your work

**Comment by senmu-a** [comment]
Thanks for your suggestion to make the documentation better and I learned a lot. :D

File: files/zh-cn/web/http/caching/index.md [hunk 6]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+200 OK HTTP/1.1
+Content-Type: application/javascript
+Content-Length: 1024
+Cache-Control: public, max-age=31536000, immutable
+Last-Modified: Tue, 22 Feb 2022 20:20:20 GMT
+ETag: YsAIAAAA-QG4G6kCMAMBAAAAAAAoK
+```
 
-![The Vary header leads cache to use more HTTP headers as key for the cache.](https://mdn.mozillademos.org/files/13769/HTTPVary.png)
+**缓存破坏** 是一种通过在内容更改时更改 URL 来使响应在很长一段时间内可缓存的技术。该技术可以应用于所有子资源，例如图像。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:591]:
```suggestion
**缓存破坏**是一种通过在内容更改时更改 URL 来使响应在很长一段时间内可缓存的技术。该技术可以应用于所有子资源，例如图像。
```

File: files/zh-cn/web/http/caching/index.md [hunk 7]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+<!doctype html>
+…
 ```
 
-#### 验证方式
+一旦响应在服务器上过期，您可能希望覆盖该响应，但是一旦存储响应，服务器就无法执行任何操作——因为由于缓存，不再有请求到达服务器。
+
+规范中提到的方法之一是使用不安全的方法（例如 POST ）发送对同一 URL 的请求，但对于许多客户端而言，通常很难故意这样做。
+
+还有一个 `Clear-Site-Data: cache` 标头和值的规范，但 [并非所有浏览器都支持它](https://groups.google.com/a/mozilla.org/g/dev-platform /c/I939w1yrTp4) — 即使使用它，它也只会影响浏览器缓存，而不会影响中间缓存。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:441]:
```suggestion
还有一个 `Clear-Site-Data: cache` 标头和值的规范，但[并非所有浏览器都支持它](https://groups.google.com/a/mozilla.org/g/dev-platform/c/I939w1yrTp4)——即使使用它，它也只会影响浏览器缓存，而不会影响中间缓存。
```

File: files/zh-cn/web/http/caching/index.md [hunk 8]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
 Cache-Control: max-age=31536000
+
+<!doctype html>
+…
 ```
 
-#### 验证方式
+一旦响应在服务器上过期，您可能希望覆盖该响应，但是一旦存储响应，服务器就无法执行任何操作——因为由于缓存，不再有请求到达服务器。
+
+规范中提到的方法之一是使用不安全的方法（例如 POST ）发送对同一 URL 的请求，但对于许多客户端而言，通常很难故意这样做。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:439]:
```suggestion
规范中提到的方法之一是使用不安全的方法（例如 POST）发送对同一 URL 的请求，但对于许多客户端而言，通常很难故意这样做。
```

File: files/zh-cn/web/http/caching/index.md [hunk 9]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+
+为了防止这种情况，`immutable` 指令可用于明确指示不需要重新验证，因为内容永远不会改变。
+
+```http
+Cache-Control: max-age=31536000, immutable
+```
+
+这可以防止在重新加载期间进行不必要的重新验证。
+
+请注意，[Chrome 已更改其实施](https://blog.chromium.org/2017/01/reload-reloaded-faster-and-leaner-page_26.html) 而不是实施该指令，因此重新验证不是在重新加载子资源期间执行。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:419]:
```suggestion
请注意，[Chrome 已更改其实现](https://blog.chromium.org/2017/01/reload-reloaded-faster-and-leaner-page_26.html)而不是实现该指令，因此重新验证不是在重新加载子资源期间执行。
```

File: files/zh-cn/web/http/caching/index.md [hunk 10]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+该行为也在 [Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) 标准中定义，并且可以通过将缓存模式设置为 `no-cache`（注意 `reload` 不是这种情况下的正确模式）：
+
+```js
+// 注意：“reload” 不是正常重新加载的正确模式； “no-cache” 才是
+fetch("/", { cache: "no-cache" });
+```
+
+### 强制重新加载
+
+出于向后兼容的原因，浏览器在重新加载期间使用 `max-age=0` —— 因为在 HTTP/1.1 之前的许多过时的实现中不理解 `no-cache`。但是在这个用例中，`no-cache` 已被支持，并且**强制重新加载**是绕过缓存响应的另一种方法。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:383]:
```suggestion
出于向后兼容的原因，浏览器在重新加载期间使用 `max-age=0`——因为在 HTTP/1.1 之前的许多过时的实现中不理解 `no-cache`。但是在这个用例中，`no-cache` 已被支持，并且**强制重新加载**是绕过缓存响应的另一种方法。
```

File: files/zh-cn/web/http/caching/index.md [hunk 11]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+（来自 Chrome、Edge 和 Firefox 的请求看起来很像上面的；来自 Safari 的请求看起来会有点不同。）
+
+请求中的 `max-age=0` 指令指定“重用 age 为 0 或更少的响应”——因此，中间存储的响应不会被重用。
+
+请求通过 `If-None-Match` 和 `If-Modified-Since` 进行验证。
+
+该行为也在 [Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) 标准中定义，并且可以通过将缓存模式设置为 `no-cache`（注意 `reload` 不是这种情况下的正确模式）：
+
+```js
+// 注意：“reload” 不是正常重新加载的正确模式； “no-cache” 才是
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:377]:
```suggestion
// 注意：“reload”不是正常重新加载的正确模式；“no-cache”才是
```

File: files/zh-cn/web/http/caching/index.md [hunk 12]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+
+但是，不建议随意授予 `no-store`，因为你失去了 HTTP 和浏览器所拥有的许多优势，包括浏览器的后退/前进缓存。
+
+因此，要获得 Web 平台的全部功能集的优势，最好将 `no-cache` 与 `private` 结合使用。
+
+## 重新加载和强制重新加载
+
+可以对请求和响应执行验证。
+
+**重新加载** 和 **强制重新加载** 操作是从浏览器端执行验证的常见示例。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:352]:
```suggestion
**重新加载**和**强制重新加载**操作是从浏览器端执行验证的常见示例。
```

File: files/zh-cn/web/http/caching/index.md [hunk 13]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
 
 ```http
+Cache-Control: no-cache
+```
+
+如果服务端不支持条件请求，你可以强制客户端每次都访问服务端，总是得到最新的 200 OK 响应。
+
+### 兼容过时的实现
+
+作为忽略`no-store`的过时实现的解决方法，你可能会看到使用了诸如以下内容的 kitchen-sink 标头：
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:326]:
```suggestion
作为忽略 `no-store` 的过时实现的解决方法，你可能会看到使用了诸如以下内容的 kitchen-sink 标头：
```

File: files/zh-cn/web/http/caching/index.md [hunk 14]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+<!doctype html>
+…
+```
+
+存储示例响应的缓存计算响应生成后经过的时间，并将结果用作响应的 _age_ 。
+
+对于示例响应，`max-age` 的含义如下：
+
+- 如果响应的 age _小于_ 一周，则响应为 _fresh_。
+- 如果响应的 age _超过_ 一周，则响应为 _stale_。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:119]:
```suggestion
- 如果响应的 age *小于*一周，则响应为 _fresh_。
- 如果响应的 age *超过*一周，则响应为 _stale_。
```

File: files/zh-cn/web/http/caching/index.md [hunk 15]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+…
+```
+
+试探性地知道，整整一年没有更新的内容在那之后的一段时间内不会更新。因此，客户端存储此响应（尽管缺少 `max-age`）并重用它一段时间。复用多长时间取决于实现，但规范建议存储后大约 10%（在本例中为 0.1 年）的时间。
+
+启发式缓存是在 `Cache-Control` 被广泛采用之前出现的一种解决方法，基本上所有响应都应明确指定 `Cache-Control` 标头。
+
+## 基于 age 的缓存策略
+
+存储的 HTTP 响应有两种状态：**fresh** 和 **stale**。 _fresh_ 状态通常表示响应仍然有效，可以重复使用，而 _stale_ 状态表示缓存的响应已经过期。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:98]:
```suggestion
存储的 HTTP 响应有两种状态：**fresh** 和 **stale**。_fresh_ 状态通常表示响应仍然有效，可以重复使用，而 _stale_ 状态表示缓存的响应已经过期。
```

File: files/zh-cn/web/http/caching/index.md [hunk 16]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
-**HTTP/1.1**定义的 {{HTTPHeader("Cache-Control")}} 头用来区分对缓存机制的支持情况，请求头和响应头都支持这个属性。通过它提供的不同的值来定义缓存策略。
+```http
+Cache-Control: no-store, no-cache, max-age=0, must-revalidate, proxy-revalidate
+```
 
-#### 没有缓存
+然而，近年来，随着 HTTPS 变得越来越普遍，客户端/服务器通信变得加密，在许多情况下，路径中的代理缓存只能传输响应而不能充当缓存。因此，在这种情况下，无需担心甚至无法看到响应的过时代理缓存的实现。
 
-缓存中不得存储任何关于客户端请求和服务端响应的内容。每次由客户端发起的请求都会下载完整的响应内容。
+另一方面，如果 {{Glossary("TLS")}} 桥接代理通过在 PC 上安装来自组织管理的 {{Glossary("Certificate_authority", "CA (certificate authority)")}} 证书，以中间人方式解密所有通信，并执行访问控制等，则可以查看响应的内容并将其缓存。但是，由于 [证书透明度（certificate transparency）](/zh-CN/docs/Web/Security/Certificate_Transparency) 在最近几年变得很普遍，并且一些浏览器只允许使用证书签署时间戳（signed certificate timestamp）颁发的证书，因此这种方法需要应用于企业策略。在这样的受控环境中，无需担心代理缓存“已过时且未更新”。
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:53]:
```suggestion
另一方面，如果 {{Glossary("TLS")}} 桥接代理通过在 PC 上安装来自组织管理的 {{Glossary("Certificate_authority", "CA")}} 证书，以中间人方式解密所有通信，并执行访问控制等，则可以查看响应的内容并将其缓存。但是，由于[证书透明度（certificate transparency）](/zh-CN/docs/Web/Security/Certificate_Transparency)在最近几年变得很普遍，并且一些浏览器只允许使用证书签署时间戳（signed certificate timestamp）颁发的证书，因此这种方法需要应用于企业策略。在这样的受控环境中，无需担心代理缓存“已过时且未更新”。
```

File: files/zh-cn/web/http/caching/index.md [hunk 17]
```diff
@@ -5,154 +5,656 @@ original_slug: Web/HTTP/Caching_FAQ
...
+可复用性有几个优点。首先，由于不需要将请求传递到源服务器，因此客户端和缓存越近，响应速度就越快。最典型的例子是浏览器本身为浏览器请求存储缓存。
+
+此外，当响应可复用时，源服务器不需要处理请求——因为它不需要解析和路由请求、根据 cookie 恢复会话、查询数据库以获取结果或渲染模板引擎。这减少了服务器上的负载。
+
+缓存的正确操作对系统的稳定运行至关重要。
 
 ## 不同种类的缓存
 
-缓存是一种保存资源副本并在下次请求时直接使用该副本的技术。当 Web 缓存发现请求的资源已经被存储，它会拦截请求，返回该资源的拷贝，而不会去源服务器重新下载。这样带来的好处有：缓解服务器端压力，提升性能 (获取资源的耗时更短了)。对于网站来说，缓存是达到高性能的重要组成部分。缓存需要合理配置，因为并不是所有资源都是永久不变的：重要的是对一个资源的缓存应截止到其下一次发生改变（即不能缓存过期的资源）。
+在 [HTTP Caching](https://httpwg.org/specs/rfc9111.html) 标准中, 有两种不同类型的缓存：**私有缓存**和**共享缓存**.
```
Comment (yin1999) [inline files/zh-cn/web/http/caching/index.md:19]:
```suggestion
在 [HTTP Caching](https://httpwg.org/specs/rfc9111.html) 标准中, 有两种不同类型的缓存：**私有缓存**和**共享缓存**。
```

**Review by yin1999 — COMMENTED** [review]
typo

**Pull request merged**